src/main/java/de/fraunhofer/aisec/codyze/config/CpgConfiguration.kt

```diff
@@ -140,14 +126,114 @@ class TranslationSettings {
     )
     var analyzeIncludes = false
 
+    @JsonIgnore @ArgGroup(exclusive = true) val includesCLI = IncludesArgGroup()
+    var includes: Array<File> = emptyArray()
+
+    @JsonIgnore @ArgGroup(exclusive = true) val enabledIncludesCLI = EnabledIncludesArgGroup()
+    var enabledIncludes: Array<File> = emptyArray()
+
+    @JsonIgnore @ArgGroup(exclusive = true) val disabledIncludesCLI = DisabledIncludesArgGroup()
+    var disabledIncludes: Array<File> = emptyArray()
+}
+
+class PassesArgGroup {
+    var append = false
+    var matched = false
+
+    var passes: List<Pass> = emptyList()
+
+    @Option(
+        names = ["--passes"],
+        paramLabel = "pass",
+        description =
+            [
+                "CPG passes in the order in which they should be executed, fully qualified name of the classes only. If default-passes is specified, the default passes are executed first."],
+        split = ","
+    )
+    fun match(passes: List<Pass?>) {
+        matched = true
+        this.passes = passes.filterNotNull()
+    }
+
+    @Option(
+        names = ["--passes+"],
+        paramLabel = "<pass>",
+        description =
+            ["See passes, but appends the values to the ones specified in configuration file."],
+        split = ","
+    )
+    fun append(value: List<Pass?>) {
+        append = true
+        match(value)
+    }
+}
+
+class SymbolsArgGroup {
+    var append = false
+    var matched = false
+
+    var symbols: Map<String, String> = emptyMap()
+
+    @Option(
+        names = ["--symbols"],
+        paramLabel = "<symbol=definition>",
+        description = ["Definition of additional symbols"],
+        split = ","
+    )
+    fun match(value: Map<String, String>) {
+        matched = true
+        this.symbols = value
+    }
+
+    @Option(
+        names = ["--symbols+"],
+        paramLabel = "<symbol=definition>",
+        description =
+            ["See --symbols, but appends the values to the ones specified in configuration file."],
+        split = ","
+    )
+    fun append(value: Map<String, String>) {
+        append = true
+        match(value)
+    }
+}
+
+class IncludesArgGroup {
+    var append = false
+    var matched = false
+
+    var includes: Array<File> = emptyArray()
+
     @Option(
         names = ["--includes"],
         description =
             [
                 "Path(s) containing include files. Path must be separated by \':\' (Mac/Linux) or \';\' (Windows)."],
         split = ":|;"
     )
-    var includes: Array<File> = emptyArray()
+    fun match(value: Array<File>) {
+        matched = true
+        this.includes = value
+    }
+
+    @Option(
+        names = ["--includes+"],
+        paramLabel = "<includes>",
+        description =
+            ["See --includes, but appends the values to the ones specified in configuration file."],
+        split = ":|;"
```

Comment:
Do we continue using `:|;` or should we switch to system agnostic path separators?